pytest_html/plugin.py

```diff
@@ -131,24 +134,27 @@ def __lt__(self, other):
                      'XPassed', 'Skipped', 'Passed')
             return order.index(self.outcome) < order.index(other.outcome)
 
-        def append_extra_html(self, extra):
+        def append_extra_html(self, extra, extra_index, test_index):
             href = None
             if extra.get('format') == extras.FORMAT_IMAGE:
-                image_src = None
                 if self.contained_html:
                     image_src = 'data:image/png;base64,{0}'.format(
                             extra.get('content'))
                     href = '#'
                 else:
-                    hash_key = self.test_id + self.outcome
+                    hash_key = self.test_id + extra_index + test_index
                     hash_generator = hashlib.md5()
                     hash_generator.update(hash_key)
-                    image_file_name = '{0}.jpg'.format(hash_generator.digest)
+                    image_file_name = '{0}.png'.format(hash_generator.digest)
+
+                    dir_name = os.path.join(os.path.dirname(self.logfile),
```

Comment:
I would still prefer to just have the image path and use that for creating any necessary directories, but I'll leave it up to you.